src/main/java/ui/Printer.java

```diff
@@ -9,6 +9,8 @@
  * This class manages the output of text.
  */
 public final class Printer {
+    private boolean isDisplayingHelpMessages = true;

```

Comment:
I feel that this variable should not be in the Printer class, and that it should be a variable in the command class which is supposed to encapsulate all info relevant to command execution. 

then in Parser you could set this variable which is now inside the command object.

Reply:
Hmm true, that seems more consistent set colour portion also